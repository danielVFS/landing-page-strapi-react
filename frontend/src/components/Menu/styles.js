import styled, { css } from 'styled-components';

import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    right: 0;
    left: 0;
    width: 100%;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};
    transition: all 300ms ease-in-out;

    > ${SectionContainer} {
      padding-bottom: 0;
      padding-top: 0;
    }

    & ${Heading} {
      margin-bottom: 0;
      margin-top: 0;
    }

    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible(theme)}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        margin: 0;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.small};
        display: flex;
        justify-content: center;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: ${theme.spacings.xsmall};

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.small} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 6;
    top: 2rem;
    right: 2rem;
    width: 4rem;
    height: 4rem;
    border: none;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: none;
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }

    > svg {
      width: 3rem;
      height: 3rem;
    }
  `}
`;
