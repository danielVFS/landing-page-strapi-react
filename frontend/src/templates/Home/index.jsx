import { useEffect, useState } from 'react';

import * as Styled from './styles';
import { BaseTemplate } from '../BaseTemplate';
import { mockBase } from '../BaseTemplate/mock';
import { mapData } from '../../api/map-data';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../Loading';

function Home() {
  const [data, setData] = useState([]);

  useEffect(() => {
    const load = async () => {
      try {
        const data = await fetch(
          'http://localhost:1337/pages/?slug=landing-page',
        );
        const json = await data.json();
        const pageData = mapData(json);

        setData(pageData[0]);
      } catch (error) {
        setData(undefined);
      }
    };

    load();
  }, []);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  return <BaseTemplate {...mockBase} />;
}

export default Home;
