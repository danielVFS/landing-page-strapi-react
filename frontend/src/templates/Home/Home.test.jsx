import { screen } from '@testing-library/react';

import Home from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

test('render Home', () => {
  const { debug } = renderTheme(<Home />);

  const headingContainer = screen.getByRole('heading', { name: 'Hello World' })
    .parentElement;

  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  });
  expect(headingContainer).toMatchSnapshot();
  expect(headingContainer).toHaveStyleRule('background', 'red');
});
