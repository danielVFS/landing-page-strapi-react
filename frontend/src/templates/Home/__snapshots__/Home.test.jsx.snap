// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home 1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    Hello World
  </h1>
</div>
`;
